src/main/scala/is/hail/rvd/OrderedRVD.scala

```diff
@@ -718,19 +707,12 @@ object OrderedRVD {
         def next(): RegionValue = {
           val rv = it.next()
 
-          if (i < partitioner.rangeBounds.length) {
-            assert(typ.pkRowOrd.compare(
-              partitioner.region, partitioner.loadElement(i),
-              rv) >= 0)
-          }
-          if (i > 0)
-            assert(typ.pkRowOrd.compare(partitioner.region, partitioner.loadElement(i - 1),
-              rv) < 0)
+          assert(RegionValueInterval(partitioner.pkIntType, partitioner.region, partitioner.loadElement(i)).contains(typ.pkRowEOrd, rv))
```

Comment:
You can't pass a RegionValue to contains here, Interval.contains is expecting an annotation.  You either need to wrap it in an UnsafeRow or write a version of contains to work on a region value.  That latter is going to be annoying for a generic interval (which contains annotations) but might be OK on RegionValueInterval (the only argument I see to keep it around).